lib/query.js

```diff
@@ -2609,12 +2634,43 @@ Query.prototype._countDocuments = async function _countDocuments() {
   applyGlobalMaxTimeMS(this.options, this.model);
   applyGlobalDiskUse(this.options, this.model);
 
-  const conds = this._conditions;
   const options = this._optionsForExec();
 
+  this._applyTranslateAliases(options);
+
+  const conds = this._conditions;
+
   return this._collection.collection.countDocuments(conds, options);
 };
 
+/*!
+ * If `translateAliases` option is set, call `Model.translateAliases()`
+ * on the following query properties: filter, projection, update, distinct.
+ */
+
+Query.prototype._applyTranslateAliases = function _applyTranslateAliases(options) {
+  let applyTranslateAliases = false;
+  if ('translateAliases' in this._mongooseOptions) {
+    applyTranslateAliases = this._mongooseOptions.translateAliases;
+  } else if (this?.model?.schema?._userProvidedOptions?.translateAliases != null) {
+    applyTranslateAliases = this.model.schema._userProvidedOptions.translateAliases;
+  } else if (this?.model?.base?.options?.translateAliases != null) {
```

Comment:
```suggestion
  } else if (this.model?.schema?._userProvidedOptions?.translateAliases != null) {
    applyTranslateAliases = this.model.schema._userProvidedOptions.translateAliases;
  } else if (this.model?.base?.options?.translateAliases != null) {
```

`this` shouldnt be undefined here, so it should not need optional chaining, otherwise it would need to be added in the first `if` too